[tempo-distributed] Upgrade to Tempo 2.4.0

faustodavid opened the pull request:
Upgrade Tempo to 2.4.0 and use the new multi-cache config. 

I have assigned the following roles to the memcache:
```yaml
- roles:
    - parquet-footer
    - parquet-column-idx
    - parquet-offset-idx
    - bloom
    - frontend-search
 ```

joe-elliott commented:
this lgtm.  to be clear i'm not sure that `parquet-column-idx` and `parquet-offset-idx` function correctly. i thought i removed them from the tempo docs, but it seems they are still there.

if anyone wants to experiment with these roles to see if tempo does, in fact, use them please do and report back.

joe-elliott commented on charts/tempo-distributed/values.yaml:
```diff
@@ -1263,6 +1263,21 @@ config: |
+  cache:
+    caches:
+      - roles:
+          - parquet-footer
```
missing `parquet-page` but this might be for the best. `parquet-page` is incredibly high volume and would forcibly flush out the other values.

faustodavid replied:
I saw the warning in the docs, that is why I didn't want to add it as the default.

joe-elliott reviewed (approved):
for reference we run 3 memcached clusters internally:

1) `frontend-search` - small cluster
2) `parquet-footer`, `parquet-bloom` - medium cluster
3) `parquet-pages` - large cluster

faustodavid commented:
> this lgtm. to be clear i'm not sure that `parquet-column-idx` and `parquet-offset-idx` function correctly. i thought i removed them from the tempo docs, but it seems they are still there.
> 
> if anyone wants to experiment with these roles to see if tempo does, in fact, use them please do and report back.



> for reference we run 3 memcached clusters internally:
> 
> 1. `frontend-search` - small cluster
> 2. `parquet-footer`, `parquet-bloom` - medium cluster
> 3. `parquet-pages` - large cluster

Hey @joe-elliott, I remove `parquet-column-idx` and `parquet-offset-idx` from this PR, but I will test them in our cluster and let the team know if we see some improvements. I have also refactored this PR to make the cache configurable and added a comment pointing to the docs. Can you review it again, please?

edgarkz commented:
@faustodavid 
Any chance for merging soon? its already 2 weeks since 2.4 release
Thank you

faustodavid commented:
> @faustodavid Any chance for merging soon? its already 2 weeks since 2.4 release Thank you

Hi @edgarkz , after @joe-elliott comments I changed the code a bit, so I am waiting for their review.

CC. @zanhsieh

joe-elliott commented:
Thanks for the ping. Apologies for missing the first. I'll see if i can get a second +1 from someone internally.

@faustodavid this is great work, thank you!

mapno reviewed (approved):
LGTM, thanks!

z0rc commented:
This is breaking change for anyone using external memcached. It should be a major release with migration instructions.
